spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/handler/predicate/PathRoutePredicateFactory.java

```diff
@@ -82,7 +88,15 @@ public Predicate<ServerWebExchange> apply(Config config) {
 		synchronized (this.pathPatternParser) {
 			pathPatternParser.setMatchOptionalTrailingSeparator(config.isMatchTrailingSlash());
 			config.getPatterns().forEach(pattern -> {
-				PathPattern pathPattern = this.pathPatternParser.parse(pattern);
+				String basePath = webFluxProperties.getBasePath();
+				boolean basePathIsNotBlank = StringUtils.hasText(basePath);
+				if (basePathIsNotBlank) {
+					if (pattern.length() > 1 && !pattern.startsWith("/")) {
+						basePath += ("/");
+					}
+				}
+				String pathPatternStr = basePathIsNotBlank ? basePath + pattern : pattern;
```

Comment:
why not place this in previous `if` statement where we know `basePathIsNotBlank == true`